fault-proof/src/config.rs

```diff
@@ -242,3 +255,199 @@ pub struct FaultDisputeGameConfig {
     pub use_sp1_mock_verifier: bool,
     pub verifier_address: String,
 }
+
+/// How many chunks the range proof input is partitioned into (1-16 inclusive).
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub struct RangeSplitCount(NonZeroU8);
+
+impl RangeSplitCount {
+    pub const MAX: u8 = 16;
+
+    /// Create a new `RangeSplitCount`.
+    pub fn new(count: u8) -> Result<Self> {
+        if count == 0 || count > Self::MAX {
+            bail!("range splits must be between 1 and 16, got {count}");
+        }
+
+        let count = NonZeroU8::new(count)
+            .ok_or_else(|| anyhow::anyhow!("range splits must be non zero"))?;
+
+        Ok(Self(count))
+    }
+
+    /// Returns a `RangeSplitCount` of one.
+    pub fn one() -> Self {
+        Self(NonZeroU8::new(1).expect("1 is non-zero"))
+    }
+
+    /// Convert to `usize`.
+    pub fn to_usize(self) -> usize {
+        self.0.get() as usize
+    }
+
+    /// Split `[start, end)` into up to `count` contiguous, non-empty subranges.
+    ///
+    /// Behavior:
+    /// - Errors if `start > end` or the range is empty.
+    /// - Caps the number of produced segments to the number of blocks in the range.
+    /// - Uses ceil division to keep segments as even as possible; the final segment takes any
+    ///   remainder.
+    /// - Always returns ranges that exactly cover `[start, end)` with no gaps or overlaps.
+    pub fn split(&self, start: u64, end: u64) -> Result<Vec<(u64, u64)>> {
+        let total = end.checked_sub(start).ok_or_else(|| {
+            anyhow::anyhow!("start block {start} is greater than end block {end}")
```

Comment:
```suggestion
            anyhow::anyhow!("start block {start} is greater than or equal to end block {end}")
```

Maybe non zero here as well to handle below together?
```
        if total == 0 {
            bail!("start block equals end block ({start}); nothing to prove");
        }
```